### ISSUE TITLE
Email functions cause a Connection Refused error

### ISSUE THREAD

**Issue by bschwind** [getredash/redash#1182]
### Issue Summary

Test emails are sent successfully but actual user registration and password emails fail with Connection Refused

AWS AMI: ami-78967519
Redash Version: redash.0.10.1.b1834

### Steps to Reproduce

- Configure the .env file so that a test email from manage.py successfully sends an email
- Add a new user
- The email is not received, and this trace is logged in `logs/celery_error.log`

```
[2016-07-12 06:00:24,074: ERROR/Worker-64] redash.tasks.send_mail[bae2b7b6-297e-4af9-90df-0595034923bc]: Failed sending message: Admin invited you to join Re:dash
Traceback (most recent call last):
  File "/opt/redash/redash.0.10.1.b1834/redash/tasks/general.py", line 32, in send_mail
    mail.send(message)
  File "/usr/local/lib/python2.7/dist-packages/flask_mail.py", line 491, in send
    with self.connect() as connection:
  File "/usr/local/lib/python2.7/dist-packages/flask_mail.py", line 144, in __enter__
    self.host = self.configure_host()
  File "/usr/local/lib/python2.7/dist-packages/flask_mail.py", line 158, in configure_host
    host = smtplib.SMTP(self.mail.server, self.mail.port)
  File "/usr/lib/python2.7/smtplib.py", line 256, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python2.7/smtplib.py", line 316, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib/python2.7/smtplib.py", line 291, in _get_socket
    return socket.create_connection((host, port), timeout)
  File "/usr/lib/python2.7/socket.py", line 571, in create_connection
    raise err
error: [Errno 111] Connection refused
```

### Technical details:

* How did you install Redash: Used ami-78967519
* I'm using SendGrid as an email provider

**Comment by arikfr** [comment]
What email provider do you use to send those emails? What port number?

Most likely that it's being blocked by AWS. We use MailGun's SMTP server with a non standard port 2525 because of this.

**Comment by bschwind** [comment]
Here's my environment setup:

```
export REDASH_MAIL_SERVER="smtp.sendgrid.net" # default: localhost
export REDASH_MAIL_PORT="587" # default: 25
export REDASH_MAIL_USE_TLS="true" # default: False
export REDASH_MAIL_USE_SSL="false" # default: False
export REDASH_MAIL_USERNAME="REDACTED" # default: None
export REDASH_MAIL_PASSWORD="REDACTED" # default: None
export REDASH_MAIL_DEFAULT_SENDER="REDACTED" # Email address to send from
```

I don't think it's AWS because the test email sends correctly.

**Comment by arikfr** [comment]
Did you restart Celery after updating settings? 
If you haven't -> restart and try again.
If you have -> make sure you don't have old workers running.

**Comment by bschwind** [comment]
I suppose restarting Celery is different from this?

`sudo supervisorctl restart redash_server`

If so, then yeah I haven't restarted Celery. I'm not familiar with the tool, what's the preferred command for restarting Celery for Redash?

**Comment by arikfr** [comment]
Yes, the above command restarts only the web server. You can restart Celery with `sudo supervisorctl restart redash_celery` or just restart everything with `sudo supervisorctl restart all`.

**Comment by bschwind** [comment]
Ah, that did it! Thanks for the help!

**Comment by glarrain** [comment]
@arikfr perhps mentioning in the [setup docs](http://docs.redash.io/en/latest/setup.html#setup) the need to call `sudo supervisorctl restart all` might help. The only restart command mentioned is `sudo supervisorctl restart redash_server`, which would not be useful in this case.

**Comment by arikfr** [comment]
@glarrain I always wondered why people restart only the web server... thanks for pointing this out! Fixed in 2e5ec26be931e7defb7574d85eae9d60d024ba25.
